cmd/createcluster.go

```diff
@@ -558,27 +574,39 @@ func generateKeys(numDVs int) ([]tbls.PrivateKey, error) {
 }
 
 // createDepositDatas creates a slice of deposit datas using the provided parameters and returns it.
-func createDepositDatas(withdrawalAddresses []string, network string, secrets []tbls.PrivateKey) ([]eth2p0.DepositData, error) {
+func createDepositDatas(withdrawalAddresses []string, network string, secrets []tbls.PrivateKey, depositAmounts []eth2p0.Gwei) ([][]eth2p0.DepositData, error) {
 	if len(secrets) != len(withdrawalAddresses) {
 		return nil, errors.New("insufficient withdrawal addresses")
 	}
+	if len(depositAmounts) == 0 {
+		return nil, errors.New("deposit amounts must not be empty")
+	}
 
-	return signDepositDatas(secrets, withdrawalAddresses, network)
+	return signDepositDatas(secrets, withdrawalAddresses, network, depositAmounts)
 }
 
 // writeDepositData writes deposit data to disk for the DVs for all peers in a cluster.
-func writeDepositData(depositDatas []eth2p0.DepositData, network string, clusterDir string, numNodes int) error {
-	// Serialize the deposit data into bytes
-	bytes, err := deposit.MarshalDepositData(depositDatas, network)
-	if err != nil {
-		return err
-	}
-
-	for i := 0; i < numNodes; i++ {
-		depositPath := path.Join(nodeDir(clusterDir, i), "deposit-data.json")
-		err = os.WriteFile(depositPath, bytes, 0o400) // read-only
+func writeDepositData(depositDatas [][]eth2p0.DepositData, network string, clusterDir string, numNodes int) error {
+	// The loop across partial amounts
+	for i := range depositDatas {
+		// Serialize the deposit data into bytes
+		bytes, err := deposit.MarshalDepositData(depositDatas[i], network)
 		if err != nil {
-			return errors.Wrap(err, "write deposit data")
+			return err
+		}
+
+		filename := "deposit-data.json"
```

Comment:
I wonder if it's worth differentiating file names when there are multiple deposit amounts vs a single one.

I think it's fine keeping it the same, i.e. when there's a single deposit amount, create a file called `deposit-data-0-32eth.json`.

@dB2510 @xenowits thoughts?